alfalfa_web/server/api-v2.js

```diff
@@ -76,26 +87,119 @@ router.get(/\/$/, (req, res) => {
  *                   uploadPath: uploads/82ae8d50-f837-11ec-bda1-355419177ef9/refrig_case_osw_2.zip
  *                   errorLog: ''
  */
-router.get("/sites", async (req, res) => {
+router.get("/runs", async (req, res, next) => {
   api
-    .listSites()
-    .then((data) => res.json({ data }))
-    .catch(() => res.sendStatus(500));
+    .listRuns()
+    .then((payload) => res.json({ payload }))
+    .catch((err) => {
+      next(err);
+    });
+});
+
+router.param("runId", (req, res, next, id) => {
+  const error = validate(
```

Comment:
Whenever `runId` is a param in the URL this function will parse the id and add the `run` object to the request object. This allows all downstream function to not have to worry about validating and processing the id. This also allows me to check if an alias exists in cases where the id doesn't correspond to a run.